core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -131,6 +132,7 @@ public class RESTSessionCatalog extends BaseViewSessionCatalog
           .add(Endpoint.V1_REGISTER_TABLE)
           .add(Endpoint.V1_REPORT_METRICS)
           .add(Endpoint.V1_COMMIT_TRANSACTION)
+          .add(Endpoint.V1_TABLE_CREDENTIALS)
```

Comment:
we can't add this here, since it's not a default endpoint and that would break backwards compatibility. See also the comment in L118